prov/shm/src/smr_progress.c

```diff
@@ -513,11 +514,14 @@ static int smr_progress_ipc(struct smr_cmd *cmd, enum fi_hmem_iface iface,
 						     ptr, cmd->msg.hdr.size);
 	}
 
-	if (cmd->msg.data.ipc_info.iface == FI_HMEM_ZE)
+	if (cmd->msg.data.ipc_info.iface == FI_HMEM_ZE) {
 		close(ipc_fd);
-
-	/* Truncation error takes precedence over close_handle error */
-	ret = ofi_hmem_close_handle(cmd->msg.data.ipc_info.iface, base);
+		/* Truncation error takes precedence over close_handle error */
+		ret = ofi_hmem_close_handle(cmd->msg.data.ipc_info.iface, base);
+	} else {
+		/* Decrement the use_cnt of the ipc mr_entry so it can be cleaned properly */
+		ofi_mr_cache_delete(ep->ipc_cache, mr_entry);
+	}
```

Comment:
This flow doesn't look correct.  There shouldn't be some special flow for Ze that's different from the common flow.

Reply:
The reason for this flow is that this ipc cache is not enabled for Ze, within this PR (because I cannot test it). So I do not intend to change any existing Ze behavior. Currently Ze opens and closes handle each time in `smr_progress_ipc`. So this `ofi_mr_cache_delete` is only called for non Ze path.